src/spatialexperiment/SpatialExperiment.py

```diff
@@ -0,0 +1,814 @@
+                Slot for low-dimensionality embeddings.
+
+                Usually a dictionary with the embedding method as keys (e.g., t-SNE, UMAP)
+                and the dimensions as values.
+
+                Embeddings may be represented as a matrix or a data frame, must contain a shape.
+
+            main_experiment_name:
+                A string, specifying the main experiment name.
+
+            alternative_experiments:
+                Used to manage multi-modal experiments performed on the same sample/cells.
+
+                Alternative experiments must contain the same cells (rows) as the primary experiment.
+                It's a dictionary with keys as the names of the alternative experiments
+                (e.g., sc-atac, crispr) and values as subclasses of
+                :py:class:`~summarizedexperiment.SummarizedExperiment.SummarizedExperiment`.
+
+            row_pairs:
+                Row pairings/relationships between features.
+
+                Defaults to None.
+
+            column_pairs:
+                Column pairings/relationships between cells.
+
+                Defaults to None.
+
+            spatial_coords:
+                Optional :py:class:`~biocframe.BiocFrame.BiocFrame` containing columns of spatial coordinates. Must be the same length as `column_data`. Typical column names might include:
+
+                    - **['x', 'y']**: For simple 2D coordinates.
+                    - **['pxl_col_in_fullres', 'pxl_row_in_fullres']**: For pixel-based coordinates in full-resolution images.
+
+                Spatial coordinates are coerced to a
+                :py:class:`~biocframe.BiocFrame.BiocFrame`. Defaults to None.
+
+            img_data:
+                Optional :py:class:`~biocframe.BiocFrame.BiocFrame` containing the image data, structured with the following columns:
+                    - **sample_id** (str): A string identifier for the sample to which an image corresponds.
+                    - **image_id** (str): A unique string identifier for each image within each sample.
+                    - **data** (SpatialImage): The image itself, represented as a SpatialImage object.
+                    - **scale_factor** (float): A numerical value that indicates the scaling factor applied to the image.
+
+                Image data are coerced to a
+                :py:class:`~biocframe.BiocFrame.BiocFrame`. Defaults to None.
+
+            validate:
+                Internal use only.
+        """
+        _validate_spatial_coords(spatial_coords=spatial_coords, column_data=column_data)
+        _validate_img_data(img_data=img_data)
+
+        if column_data is None:
+            column_data = biocframe.BiocFrame({"sample_id": []})
+
+        _validate_column_data(column_data=column_data, img_data=img_data)
+
+        self._spatial_coords = _sanitize_frame(
+            spatial_coords, num_rows=column_data.shape[0]
+        )
+        self._img_data = _sanitize_frame(img_data, num_rows=0)
```

Comment:
- [ ] I would call the super to make sure all the parameters are correct. 
- [ ] Only validate the spatial coords and img data if validate is true.